contrib/renew-client-cert

```diff
@@ -0,0 +1,66 @@
+#!/bin/sh -e
+
+# This is an OnChanged handler that can handle requests to renew the client
+# certificates used by the device to connect to the device gateway
+
+[ -z "$CONFIG_FILE" ] && (echo "No CONFIG_FILE specified"; exit 1)
+[ -f "$CONFIG_FILE" ] || (echo "$CONFIG_FILE does not exist"; exit 1)
+[ -z "$SOTA_DIR" ] && (echo "No SOTA_DIR specified"; exit 1)
+[ -d "$SOTA_DIR" ] || (echo "$SOTA_DIR does not exist"; exit 1)
+
+fioconfig=$(which fioconfig)
+
+timestamp=$(cat $CONFIG_FILE | grep TIMESTAMP | cut -d= -f2)
+[ -z "$timestamp" ] && (echo "No TIMESTAMP found in config file"; exit 1)
+echo "Timestamp for rotation is $timestamp"
+
+est_server=$(cat $CONFIG_FILE | grep ESTSERVER | cut -d= -f2)
+[ -z "$est_server" ] && (echo "No ESTSERVER found in config file"; exit 1)
+echo "EST server for rotation is $est_server"
+
+# We need to toggle between two different pkcs11 slot ids to rotate on HSMs:
+pkey_ids=$(cat $CONFIG_FILE | grep PKEYIDS | cut -d= -f2)
+if [ -n "$pkey_ids" ]; then
+	echo "PKCS11 key IDs: ${pkey_ids}"
+fi
+cert_ids=$(cat $CONFIG_FILE | grep CERTIDS | cut -d= -f2)
+if [ -n "$cert_ids" ]; then
+	echo "PKCS11 cert IDs: ${cert_ids}"
+fi
+
+rotate() {
+	extra_args=""
+	if [ -n "$pkey_ids" ]; then
+		extra_args="--pkcs11-key-ids=$pkey_ids"
+	fi
+	if [ -n "$cert_ids" ]; then
+		extra_args="$extra_args --pkcs11-cert-ids=$cert_ids"
+	fi
+	exec $fioconfig -c $SOTA_DIR renew-cert $est_server "$extra_args"
```

Comment:
How exactly retry happens if this command `exec $fioconfig -c $SOTA_DIR renew-cert $est_server "$extra_args"` fails because `http post /simpleenroll` fails due to lack of network connectivity?

It seems like fioconfig runs an onchange script only if a config file/item has changed https://github.com/foundriesio/fioconfig/blob/00e77910b6778cb9fe938dadfdfd902c62f50dc3/internal/app.go#L250.